### PR TITLE
Refactor method to reduce ABC Metric size

### PR DESCRIPTION
**Pull request by ashmaroli**
Assignment Branch Condition Size for the method on repo branches (as reported by RuboCop) :
  - `master` : **29.7**
  - `PR-branch`: ~~20.1~~ **13.64**

### CODE REVIEW

File: lib/jekyll/commands/serve.rb [hunk 1]
```diff
@@ -205,27 +205,28 @@ def enable_logging(opts)
...
         def enable_ssl(opts)
-          return if !opts[:JekyllOptions]["ssl_cert"] && !opts[:JekyllOptions]["ssl_key"]
-          if !opts[:JekyllOptions]["ssl_cert"] || !opts[:JekyllOptions]["ssl_key"]
-            # rubocop:disable Style/RedundantException
-            raise RuntimeError, "--ssl-cert or --ssl-key missing."
-          end
+          jekyll_options = opts[:JekyllOptions]
+          ssl_cert       = jekyll_options["ssl_cert"]
+          ssl_key        = jekyll_options["ssl_key"]
+          return if !ssl_cert && !ssl_key
```
Comment (pathawks) [inline lib/jekyll/commands/serve.rb]:
```
return unless ssl_cert || ssl_key
```

Comment (ashmaroli) [reply lib/jekyll/commands/serve.rb]: the original line defined it along similar lines..
```ruby
return if !opts[:JekyllOptions]["ssl_cert"] && !opts[:JekyllOptions]["ssl_key"]
```
I just replaced with local variables..

Comment (pathawks) [reply lib/jekyll/commands/serve.rb]: As long as it’s being refactored, it should be refactored to be more clear :+1:

Comment (envygeeks) [reply lib/jekyll/commands/serve.rb]: You didn't make it more clear, you changed it's meaning.  It should be

```ruby
return unless ssl_cert && ssl_key
```

Comment (envygeeks) [reply lib/jekyll/commands/serve.rb]: If you want to clean it up this is what it ***should*** be

```ruby
return unless ssl_cert || ssl_key
ssl_key, ssl_cert = opts[:JekyllOptions].values_at("ssl_cert", "ssl_key")
raise RuntimeError, "missing something" unless ssl_cert && ssl_key
```

Comment (ashmaroli) [reply lib/jekyll/commands/serve.rb]: @pathawks local testing showed me that the two are functionally different even though they appear same..

Comment (envygeeks) [reply lib/jekyll/commands/serve.rb]:
```ruby
def enable_ssl(opts)
  return unless ssl_cert || ssl_key
  key, cert = opts[:JekyllOptions].values_at("ssl_cert", "ssl_key")
  raise RuntimeError, "missing something" unless cert && key

  src = Pathutil.new(opts[:JekyllOptions]["source"])
  opts[:SSLPrivateKey ] = OpenSSL::PKey::RSA.new(src.join(key).cleanpath.read)
  opts[:SSLCertificate] = OpenSSL::X509::Certificate.new(src.join(cert).cleanpath.read)
  opts[:SSLEnable] = true
end
```

Comment (ashmaroli) [reply lib/jekyll/commands/serve.rb]: ### sample file:
```ruby
def iflog(input="foo", alpha=false, beta=false)
  return if !alpha && !beta
  print "INPUT: "
  puts input
end

def unlog(input="foo", alpha=false, beta=false)
  return unless alpha && beta
  print "INPUT: "
  puts input
end
```
- load *sample* in `irb`
- exec following couplets:
  ```sh
  iflog # => nil
  unlog # => nil

  iflog("bar", true) # => INPUT: "bar"\n nil
  unlog("bar", true) # => nil

  iflog("bar", false, true) # => INPUT: "bar"\n nil
  unlog("bar", false, true) # => nil

  iflog("bar", true, true) # => INPUT: "bar"\n nil
  unlog("bar", true, true) # => INPUT: "bar"\n nil
  ```

Comment (pathawks) [reply lib/jekyll/commands/serve.rb]: I don’t get it.
By De Morgan’s Law, if we want to negate the statement, we have to negate the whole thing:

```
if !a && !b
if !!(!a && !b)
unless !(!a && !b)
unless !!a || !!b
unless a || b
```

Comment (envygeeks) [reply lib/jekyll/commands/serve.rb]: Guys, the first one checks to see if neither exists and then leaves, the second one checks to see if either one doesn't exist and then throws.  The first says if both don't exist then cool, lets bail out, the second says that if both don't exist then this is garbage.

So the first needs to be `unless a || b` and the second can be `if !a || !b` or `unless a && b`

Comment (envygeeks) [reply lib/jekyll/commands/serve.rb]: Writing it as so should make life much easier and avoid this debate later.

```
return if a.nil? && b.nil?
raise RuntimeError, "need both" unless a && b
```

Comment (envygeeks) [reply lib/jekyll/commands/serve.rb]: Who would have thought `nil?` would solve a semantics problem. :man_shrugging:

**Comment by DirtyF** [comment]
@jekyllbot: merge +dev

**Comment by ashmaroli** [comment]
Thanks for your inputs on this one @envygeeks

**Pull request merged**